File: reverse/reverse.java
import java.io.*;
import java.util.*;

public class reverse {
  public static void main(String args[]) {
    String input = "";
    System.out.println("Please enter the string to be reversed: ");
    try {
      BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
      input = br.readLine();
      String result = "";
      result = reverse(input);
      System.out.println(result);
    }
    catch(IOException e) {
      e.printStackTrace(); 
    }
  }

  public static String reverse(String input) {
    char[] output = input.toCharArray();
    String returnString = ""; 
     for (int i=output.length-1; i>=0; i--)
       returnString = returnString + output[i];
      
     return returnString;
  }
}

File: reverse/reverse.cpp
#include<iostream>
#include<string>

std::string reverse(std::string input)
{
  char array[50];
  strcpy(array,input.c_str());
  const size_t length = strlen(array);
  for(size_t i = 0; i < length/2; i++)
   std::swap(array[i],array[length-i-1]);
  std::string output(array);  
  return output;
}

int main()
{
  std::string input;
  std::cout << "Please enter the string to be reversed: ";
  std::getline(std::cin,input);
  std::string result;
  result = reverse(input);
  std::cout << "Reversed String: " <<  result << std::endl;  
  return 0;
}



